src/StorageEngine/Utils.hpp

```diff
@@ -141,8 +146,13 @@ class CinderEdge {
   bool operator==(const CinderEdge &other) const {
     return __id_ == other.__id_;
   }
+  bool operator!=(const CinderEdge &other) const {
+    return !(*this == other);
```

Comment:
```suggestion
    return this->__id_ != other.__id_;
```
See if you can work it out like this. And same for the ``CinderVertex`` class.